test_qmk_compiler.py

```diff
@@ -0,0 +1,337 @@
+        'MATRIX_ROWS': '4',
+        'MATRIX_COLS': '3',
+        'MATRIX_ROW_PINS': '{ F0, F5, F4, B4 }',
+        'MATRIX_COL_PINS': '{ F1, F7, F6 }',
+        'UNUSED_PINS': True,
+        'DIODE_DIRECTION': 'ROW2COL',
+        'DEBOUNCING_DELAY': '20',
+        'BACKLIGHT_LEVELS': '6',
+        'IS_COMMAND()': '( \\',
+        'RGB_DI_PIN': 'E6',
+        'RGBLED_NUM': '4 // Number of LEDs',
+        'RGBLIGHT_HUE_STEP': '10',
+        'RGBLIGHT_SAT_STEP': '17',
+        'RGBLIGHT_VAL_STEP': '17'
+    }
+
+
+## FIXME: Write a test for update_kb_redis.parse_rules_mk
+
+
+def test_0021_parse_rules_mk_file_cluecard():
+    """Make sure the rules.mk parsing works.
+    """
+    rules_mk = update_kb_redis.parse_rules_mk_file('qmk_firmware/keyboards/clueboard/card/rules.mk')
+    assert rules_mk == {
+        'ARCH': 'AVR8',
+        'AUDIO_ENABLE': 'yes',
+        'BACKLIGHT_ENABLE': 'yes',
+        'BLUETOOTH_ENABLE': 'no',
+        'BOOTMAGIC_ENABLE': 'no',
+        'COMMAND_ENABLE': 'yes',
+        'CONSOLE_ENABLE': 'yes',
+        'EXTRAKEY_ENABLE': 'yes',
+        'F_CPU': '16000000',
+        'F_USB': '$(F_CPU)',
+        'MCU': 'atmega32u4',
+        'MIDI_ENABLE': 'no',
+        'MOUSEKEY_ENABLE': 'yes',
+        'NKRO_ENABLE': 'no',
+        'OPT_DEFS': '-DINTERRUPT_CONTROL_ENDPOINT -DBOOTLOADER_SIZE=4096',
+        'RGBLIGHT_ENABLE': 'yes',
+        'SLEEP_LED_ENABLE': 'no',
+        'UNICODE_ENABLE': 'no'
+    }
+
+
+def test_0022_default_key():
+    """Test update_kb_redis.default_key().
+    """
+    new_key1 = update_kb_redis.default_key()
+    new_key2 = update_kb_redis.default_key()
+    assert new_key1 == {'x': 3, 'y': 4, 'w': 1}
+    assert new_key2 == {'x': 4, 'y': 4, 'w': 1}
+
+
+def test_0022_default_key_label():
+    """Test update_kb_redis.default_key('label').
+    """
+    new_key1 = update_kb_redis.default_key('label')
+    new_key2 = update_kb_redis.default_key('label')
+    assert new_key1 == {'label': 'label', 'x': 5, 'y': 4, 'w': 1}
+    assert new_key2 == {'label': 'label', 'x': 6, 'y': 4, 'w': 1}
+
+
+def test_0023_preprocess_source_cluecard():
+    """Test the clang preprocessor function.
+    """
+    keymap_text = update_kb_redis.preprocess_source('qmk_firmware/keyboards/clueboard/card/keymaps/default/keymap.c')
+    assert 'constuint16_tPROGMEMkeymaps' in keymap_text
+    assert 'RGB_TOG,RGB_SAI,RGB_VAI,RGB_HUD,RGB_HUI,RGB_MOD,RGB_SAD,RGB_VAD,BL_STEP' in keymap_text
+
+
+def test_0024_populate_enums_planck():
+    """Test the enum extraction code.
+    """
+    keymap_file = 'qmk_firmware/keyboards/planck/keymaps/default/keymap.c'
+    keymap_text = update_kb_redis.preprocess_source(keymap_file)
+    keymap = update_kb_redis.extract_layouts(keymap_text, keymap_file)
+    keymap_enums = update_kb_redis.populate_enums(keymap_text, keymap)
+    assert keymap_enums  ## FIXME I'm honestly not sure what update_kb_redis.populate_enums is for. Need to figure that out before I can test it.
```

Comment:
good to have a placeholder. is there a way to put in a warning test that doesn't trigger an error?

Reply:
Yeah, I can mark that one skipped for now. Shows up like this:

```
================================ test session starts ================================
platform darwin -- Python 3.7.0, pytest-4.0.2, py-1.7.0, pluggy-0.8.0
rootdir: /Users/zwhite/qmk_compiler, inifile: pytest.ini
collected 26 items

test_qmk_compiler.py .................s........                               [100%]

======================= 25 passed, 1 skipped in 26.74 seconds =======================
```